src/main.rs

```diff
@@ -528,6 +518,17 @@ fn get_branch_sha(
     }
 }
 
+#[get("/heads/<branch_type>/<branch_name>?<remote_name>")]
+fn get_qualified_branch_sha(
+    remote_name: Option<String>,
+    branch_name: String,
+    branch_type: String,
+    state: State<ServerState>,
+) -> Result<Json<ShaResponse>, Status> {
+    let branch_name = format!("{}/{}", branch_type, branch_name);
```

Comment:
Won't this only handle a single forward slash in the branch name?
Could we do something like

```
#[get("/heads/<branch_name..>?<remote_name>")]
```

in the `get_branch_sha` method, and then join the branch name components

Reply:
Good find!! Updated the logic.